.github/workflows/macos_ui_tests.yml

```diff
@@ -55,6 +69,8 @@ jobs:
     steps:
       - name: Check out repository
         uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.branch || github.sha }}
```

Comment:
This is required everywhere because of the new optional `branch` input. The `github.sha` is the default value when not provided.